### PR TITLE
Make the registration responsible to know if it is renewable

### PR DESCRIPTION
**Pull request by cintamani**
This is a little refactoring on the action links helpers for renewal links.
First, we drop the check on the class name as the `can?` permission check will just fail unless the object is of the right class.
Second, we give responsibility to check if a registration is renewable to the registration itself.

### CODE REVIEW

File: app/models/waste_exemptions_engine/registration.rb [hunk 1]
```diff
@@ -29,5 +33,13 @@ def state
 
       "ceased"
     end
+
+    def renewable?
```
Comment (Cruikshanks) [inline app/models/waste_exemptions_engine/registration.rb:37]: Though minor these new methods are public. So do you think we should have some unit tests for them?

Comment (cintamani) [reply app/models/waste_exemptions_engine/registration.rb:37]: Sure! will add some

File: spec/models/waste_exemptions_engine/registration_spec.rb [hunk 2]
```diff
@@ -45,45 +19,62 @@
...
+  end
+
+  describe "#in_renewable_state?" do
+    before do
+      allow(registration).to receive(:state).and_return(state)
+    end
 
-    context "when the renewal period has started" do
-      let(:expires_on) { 10.days.from_now }
+    context "whent the state is active" do
```
Comment (irisfaraway) [inline spec/models/waste_exemptions_engine/registration_spec.rb]:
```suggestion
    context "when the state is active" do
```

File: spec/models/waste_exemptions_engine/registration_spec.rb [hunk 3]
```diff
@@ -45,45 +19,62 @@
...
+    context "when the registration is not in a renewal window" do
+      it "returns false" do
+        allow(registration).to receive(:in_renewal_window?).and_return(false)
+        allow(registration).to receive(:in_renewable_state?).and_return(true)
+
+        expect(registration).to_not be_renewable
+      end
+    end
 
+    context "when the registratiion is not in a renewal state" do
```
Comment (irisfaraway) [inline spec/models/waste_exemptions_engine/registration_spec.rb]:
```suggestion
    context "when the registration is not in a renewal state" do
```

**Review by irisfaraway — CHANGES_REQUESTED** [review]
Looks good :) Just some typos!

**Review by irisfaraway — APPROVED** [review]

**Pull request merged**